docs/api/specs/webConnectionProtocol.md

```diff
@@ -0,0 +1,384 @@
+      timestamp: "2024-09-09T11:44:39+00:00"
+    }
+  };
+  candidate.postMessage(hello, { targetOrigin: "*" });
+  ```
+
+  Note that the `targetOrigin` is set to `*` as the origin of the Desktop Agent is not known at this point.
+  3. Accept the first correct response received from a candidate. Correct responses MUST correspond to either the [`WCP2LoadUrl`](https://fdc3.finos.org/schemas/next/api/WCP2LoadUrl.schema.json) or [`WCP3Handshake`](https://fdc3.finos.org/schemas/next/api/WCP3Handshake.schema.json) message schemas and MUST quote the same `meta.connectionAttemptUuid` value provided in the original `WCP1Hello` message. Stop the timeout when a correct response is received. If no response is received from any candidate, the `getAgent()` implementation MAY retry sending the `WCP1Hello` message periodically until the timeout is reached.
+  4. If a [`WCP3Handshake`](https://fdc3.finos.org/schemas/next/api/WCP3Handshake.schema.json) was received in the previous step, skip this this step and move on to 5. However, If a [`WCP2LoadUrl`](https://fdc3.finos.org/schemas/next/api/WCP2LoadUrl.schema.json) was received in the previous step:
+      - Create a hidden iframe within the page, set its URL to the URL provided by the `payload.iframeUrl` field of the message and add a handler to run when the iframe has loaded:
+          ```ts
+          const loadIframe = (url: string, loadedHandler: () => void): WindowProxy => {
+            const ifrm = document.createElement("iframe");
+            iframe.onload = loadedHandler;
+            ifrm.src = url;
+            ifrm.style.width = "0";
+            ifrm.style.height = "0";
+            ifrm.style.visibility = "0";
+            ifrm.ariaHidden = "true";
+            document.body.appendChild(ifrm);
+            return ifrm.contentWindow;
+          };
+          ```
+      - Once the frame has loaded (i.e. when the `loadedHandler` in the above example runs), repeat steps 1-3 above substituting the  iframe's `contentWindow` for the candidate window objects before proceeding to step 5. A new timeout should be used to limit the amount of time that the `getAgent()` implementation waits for a response. If the event that this subsequent timeout is exceeded, reject Error with the `ErrorOnConnect` message from the [`AgentError`](../ref/Errors#agenterror) enumeration.
+
+        :::tip
+
+        To ensure that the iframe is ready to receive the `WCP1Hello` message when the `load` event fires, implementations should call `window.addEventListener` for the `message` event synchronously and as early as possible.
+
+        :::
+
+  5. At this stage, a [`WCP3Handshake`](https://fdc3.finos.org/schemas/next/api/WCP3Handshake.schema.json) message should have be received from either a candidate parent or a hidden iframe created in 4 above. This message MUST have a `MessagePort` appended to it, which is used for further communication with the Desktop Agent.
```

Comment:
"should have be received" should be "should have been received"

Reply:
fixed in 5e8ba8b01ab5ecf86081172ff87369b1faf28540